src/transform/plugins/imsize/plugin.ts

```diff
@@ -206,6 +212,32 @@ export const imageWithSize = (md: MarkdownIt): ParserInline.RuleInline => {
             if (height !== '') {
                 token.attrs.push([ImsizeAttr.Height, height]);
             }
+
+            if (opts?.inlineSizeStyling) {
+                let style: string | undefined = '';
+
+                const widthWithPercent = width.includes('%');
+                const heightWithPercent = height.includes('%');
+
+                if (width !== '') {
+                    const widthString = widthWithPercent ? width : `${width}px`;
+                    style += `width: ${widthString};`;
+                }
+
+                if (height !== '') {
+                    if (width !== '' && !heightWithPercent && !widthWithPercent) {
+                        style += `aspect-ratio: ${width} / ${height};height: auto;`;
+                        forcedSanitizeWhiteList['aspect-ratio'] = true;
```

Comment:
We need to enforce the 'aspect-ratio' rule for inline CSS sanitizing. Otherwise, this rule may be ignored and some artifacts may occur.